packages/client/components/TeamPromptMeeting.tsx

```diff
@@ -99,26 +88,31 @@ const TeamPromptMeeting = (props: Props) => {
 
   const phase = getPhaseByTypename(phases, 'TeamPromptResponsesPhase')
   const stages = useMemo(() => {
-    const allStages = phase.stages
-      .map((stage) => {
-        return {
-          ...stage,
-          key: stage.id
-        }
-      })
-      .filter(isNotNull)
-
-    // Find the viewer's card and put it at the beginning.
+    const allStages = phase.stages.filter(isNotNull)
+
+    const nonViewerStages = allStages.filter((stage) => stage.teamMember.userId !== viewerId)
+    const orderedNonEmptyStages = nonViewerStages
+      .filter((stage) => !!stage.response?.plaintextContent)
+      .sort((stageA, stageB) =>
+        sortByISO8601Date(stageA.response!.createdAt, stageB.response!.createdAt)
+      )
+    // Empty stages are implicitly ordered by stage creation time on the backend.
+    const orderedEmptyStages = nonViewerStages.filter((stage) => !stage.response?.plaintextContent)
+    let orderedStages = [...orderedNonEmptyStages, ...orderedEmptyStages]
+
+    // Find the viewer's card.
     const viewerCardIndex = allStages.findIndex((stage) => stage.teamMember.userId === viewerId)
```

Comment:
+1 you could use `const viewerCard = allStages.find(...` to omit the index

Reply:
Nice catch!